src/components/inputs/rich-text-input/rich-text-input.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import requiredIf from 'react-required-if';
+import Html from 'slate-html-serializer';
+import Types from 'slate-prop-types';
+import { Editor as SlateEditor } from 'slate-react';
+import { RenderMarkPlugin, RenderBlockPlugin } from './plugins';
+import UndoPlugin from './plugins/undo';
+import RedoPlugin from './plugins/redo';
+import ListPlugin from './plugins/list';
+import MarkPlugin from './plugins/mark';
+import Editor from './editor';
+import rules from './utils/rules';
+
+// Create a new serializer instance with our `rules` from above.
+const html = new Html({ rules });
+
+const plugins = [
+  MarkPlugin({
+    typeName: 'bold',
+    hotkey: 'mod+b',
+    command: 'toggleBoldMark',
+    query: 'hasBoldMark',
+    // eslint-disable-next-line react/display-name, react/prop-types
+    RenderMark: ({ children, ...attributes }) => (
+      <strong {...attributes}>{children}</strong>
+    ),
+  }),
+  MarkPlugin({
+    typeName: 'italic',
+    hotkey: 'mod+i',
+    command: 'toggleItalicMark',
+    query: 'hasItalicMark',
+    // eslint-disable-next-line react/display-name, react/prop-types
+    RenderMark: props => <em {...props}>{props.children}</em>,
+  }),
+  MarkPlugin({
+    typeName: 'underlined',
+    hotkey: 'mod+u',
+    command: 'toggleUnderlinedMark',
+    query: 'hasUnderlinedMark',
+    // eslint-disable-next-line react/display-name, react/prop-types
+    RenderMark: props => <u {...props}>{props.children}</u>,
+  }),
+  RenderMarkPlugin(),
+  RenderBlockPlugin(),
+  UndoPlugin(),
+  RedoPlugin(),
+  ListPlugin({
+    typeName: 'numbered-list',
+    query: 'hasNumberedListBlock',
+    command: 'toggleNumberedListBlock',
+  }),
+  ListPlugin({
+    typeName: 'bulleted-list',
+    query: 'hasBulletedListBlock',
+    command: 'toggleBulletedListBlock',
+  }),
+];
+
+const RichTextInput = props => {
+  const { onChange } = props;
+  const onValueChange = React.useCallback(
+    ({ value }) => {
+      onChange(value);
+    },
+    [onChange]
+  );
+
+  return (
+    <SlateEditor
+      {...props}
```

Comment:
This is going to pass some props that `SlateEditor` doesn't recognize, such as `isDisabled` (which is passed below to `disabled`) or `horizontalConstraint` (which otherwise isn't being used), so if you want to spread the props you should destructure them as you did with `onChange`, or maybe have a prop-getter.

Reply:
I think I can just remove the spreading anyways